typing/types.mli

```diff
@@ -233,6 +233,16 @@ and commutable =
   | Cunknown
   | Clink of commutable ref
 
+module Internal : sig
+  type type_expr_internal =
+      { mutable desc: type_desc;
+        mutable level: int;
+        mutable scope: int;
+        id: int }
+  val lock : type_expr_internal -> type_expr
+  val unlock : type_expr -> type_expr_internal
+end
```

Comment:
Rather than exposing the actual type and conversions, why not expose the three setters and a constructor function? Those are all the operations that you are actually exposing, and it seems better to just use them directly.

Reply:
Explained above. This is not an abstract API, but rather a gaping hole to allow `Btype` to do its job.
Ideally all the functions of `Btype` would be in `Types`, and this wouldn't be needed.

Reply:
By the way, we are working also on the next step, where `type_expr` would be abstract rather than private, and constructor and access functions would be provided. But this is not in the scope of this PR, which only restricts mutation.

Reply:
I appreciate that it is a backdoor rather than the intended interface -- and a scarier name would be great too -- but that doesn't mean that the backdoor shouldn't itself have a well-designed interface. As it is the code in Btypes is being made harder to read than it needs to be.